modules/test/playwright/tests/frontend-data-set-views-web/pages/data_set/tabs/FiltersPage.ts

```diff
@@ -67,6 +99,54 @@ export class FiltersPage {
 		await this.dataSetPage.selectTab('Filters');
 	}
 
+	async assertFiltersTableRowCount(rowCount: number) {
+		await expect(
+			this.filterTable.locator('tbody').locator('tr')
+		).toHaveCount(rowCount);
+	}
+
+	async assertValidationError(text: string) {
+		const visualFeedback = this.newFilterModal.formFeedback.filter(
+			{ has: this.page.locator('.form-feedback-indicator')}
+		).first();
+
+		await expect(visualFeedback).toBeVisible();
+
+		await expect(visualFeedback).toContainText(text);
+	}
+
+	async cancelAddFilterModal() {
+		await this.newFilterModal.cancelButton.click();
+	}
+
+	async createDateRangeFilter({
+		filterBy,
+		name,
+		from,
+		to,
+	}: IDateRangeFilter) {
+		await this.openNewFilterModal({
+			dropdownItemLabel: 'Date Range',
+		});
+
+		await this.newDateRangeFilterModal.nameInput.click();
+		await this.newDateRangeFilterModal.nameInput.fill(name);
+
+		await this.newDateRangeFilterModal.filterBySelect.click();
+
+		const dateFilterOption = this.page.getByRole('option', {
+			name: filterBy,
+		});
+
+		await dateFilterOption.click();
+
+		from && await this.newDateRangeFilterModal.fromInput.fill(from);
+
+		to && await this.newDateRangeFilterModal.toInput.fill(to);
```

Comment:
It looks odd to have an expression disguising as a statement. It technically works, but it doesn't _feel_ right. It makes me think of programming patterns I've seen in Perl, that take advantage of the fact that functions there return last evaluated value :smile: 

`if(from) {await ...}` is much cleaner.